File: app/src/main/java/com/musicplayer/OpenMusic/data/base/AlbumDao.java
package com.musicplayer.OpenMusic.data.base;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.musicplayer.OpenMusic.data.Album;
import com.musicplayer.OpenMusic.data.Song;

import java.util.List;

@Dao
public interface AlbumDao {
    @Insert
    void insert(Album album);

    @Query("SELECT * FROM ALBUM")
    List<Album> getAll();

    @Query("SELECT * FROM Song WHERE Song.album_id=:albumId")
    List<Song> getSongs(String albumId);

    @Delete
    void delete(Album album);
}

File: app/src/main/java/com/musicplayer/OpenMusic/data/base/AlbumDao.kt
package com.musicplayer.OpenMusic.data.base

import androidx.room.Dao
import com.musicplayer.OpenMusic.data.Album
import com.musicplayer.OpenMusic.data.Song
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface AlbumDao {
    @Insert
    fun insert(album: Album)

    @get:Query("SELECT * FROM ALBUM")
    val all: ArrayList<Album>

    @Query("SELECT * FROM Song WHERE Song.album_id=:albumId")
    fun getSongs(albumId: String): ArrayList<Song>

    @Delete
    fun delete(album: Album)
}